src/core/socialprovider.js

```diff
@@ -167,7 +175,6 @@ XMPPSocialProvider.prototype.connect = function(continuation) {
 
     if (this.client) {
       this.client.end();
-      delete this.client;
```

Comment:
This block is confusing me now. Error events can happen at any time, not just as a result of login. So we should decide to do 1 of 3 things:
a. Do a full cleanup if an error event happens and log out.
b. Generally ignore the error
c. Figure out which errors can be returned and appropriatedly do either a or b depending on the severity.

But just ending the session without updating the appropriate local state seems like the wrong thing to do?